AB#9007 & AB#9009

andrewamondsen opened the pull request:
- Added two header parameters to the payment end point to be able to send payment origin and request origin.

andrewamondsen committed:
[AB#9007 and AB#9009]
- Added two header parameters to the payment end point to be able to send payment origin and request origin.

andrewamondsen committed:
[AB#9007 and AB#9009]
- The parameters are fixed and not provided by the user.

andrewamondsen committed:
[AB#9007 and AB#9009]
- The parameters are fixed and not provided by the user.

kevinbroeren commented on src/cashflows.js:
```diff
@@ -317,7 +317,7 @@ export function Cashflows(intentToken, isIntegration) {
 			// Only continue when the checkoutIntentPromise resolves and thus intent has been validated.
 			self._checkoutIntentPromises.push(() =>
-				self._apiRequest('post', 'payment/apple-pay/get-payment-request?token=' + self._intentToken)
+				self._apiRequest('post', 'payment/apple-pay/get-payment-request?paymentDataOrigin=cashflows-clientlib-js&token=' + self._intentToken)
```
In the CreatePaymentJob mehtod in the PaymentJobs controller in our backend you've added this parameter as a header (which has my preference), whereas here you are adding it as a query parameter to the request.  Both need to be aligned.'

I'd also instead include this parameter to the 'startPayment' method (line 526). This method is used by all payment methods, not just ApplePay and GooglePay and is called when the payment is started, as opposed to when the embedded frames payment page is loaded. We'll also need to add this parameter to the CreatePaymentFromIntent method in the PaymentIntents controller in the backend.

kevinbroeren commented on src/cashflows.js:
```diff
@@ -395,7 +395,7 @@ export function Cashflows(intentToken, isIntegration) {
 					// Only continue when the checkoutIntentPromise resolves and thus intent has been validated.
 					self._checkoutIntentPromises.push(() =>
-						self._apiRequest('post', 'payment/google-pay/get-payment-data-request?domain=' + encodeURIComponent(window.location.hostname) + '&token=' + self._intentToken + merchantIdParameter)
+						self._apiRequest('post', 'payment/google-pay/get-payment-data-request?domain=' + encodeURIComponent(window.location.hostname) + '&paymentDataOrigin=cashflows-clientlib-js&token=' + self._intentToken + merchantIdParameter)
```
Same as previous comment :)

andrewamondsen committed:
[AB#9007, AB#9009 and AB#9070]
- The PaymentDataOrigin is only sent to the CreatePaymentFromIntent end point and it's part of the request body, not a query parameter.

jgeuijen-cashflows committed:
undo PaymentDataOrigin

jgeuijen-cashflows committed:
RequestOrigin example

jgeuijen-cashflows committed:
whitespace
